docs/source/overview.rst

```diff
@@ -63,7 +63,47 @@ Main concepts
 Deploy
 ^^^^^^
 
-will be added soon
+
+Deployment is designed in the way, that user pays for himself. To do that, we use counterfactual deployment. (``create2`` function to deploy contract)
+
+The process looks like follows:
+
+- **computing contract address** - SDK computes deterministic contract address. Contract address is unique and is signed to the users public key.
+
+- **waiting for balance** - User needs to send funds on this address. He can transfer ether on his own or use on-ramp provider. SDK wait for future contract address balance change. If SDK discovers, that required funds are on this balance, sends DEPLOY request to relayer.
+
+- **deploy** - Relayer deploys the contract and immediately gets the refund from the contract.
+
+
+Deploy in-depth
+^^^^^^^^^^^^^^^
+
+
+SDK create deploy. Deploy contains the following parameters:
+
+.. image:: ../modeling/img/concepts/Deployment.png
+
+- **ensName** - ENS name chosen by user. It is the only parameter provided by the user.
+- **publicKey** - the public key of newly generated key pair on a users device.
+- **ensAddress** - address of ENS contract. It is required to properly register the ENS name.
+- **resolverAddress** - address of Resolver contract. It is required to properly register the ENS name.
+- **registrarAddress** - address of Registrar contract. It is required to properly register the ENS name.
+- **gasPrice** - gas price used in the refund.
+- **signature** - the signature of the arguments above. Signature prevents against unproperly ENS name registration and potential vulnerabilities like switch public key or gas price.
```

Comment:
the signature of all of the arguments above. Signature ensures parameters comes from the owner of the private key (paired to the public key). In particular, it prevents against malicious ENS name registration and gas price replacement.
()